accounts-db/src/accounts_index_storage.rs

```diff
@@ -118,8 +119,7 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> AccountsIndexStorage<
     ///      also creates some additional bg threads to facilitate flushing to disk asap
     /// startup=false is 'normal' operation
     pub fn set_startup(&self, startup: Startup) {
-        let value = !matches!(startup, Startup::Normal);
```

Comment:
man how does stuff like this pass review in the first place? :joy:

Reply:
https://github.com/anza-xyz/agave/blob/a1c1227145b6e8377384a5799d0a85bc4ae030c5/accounts-db/src/accounts_db.rs#L8136

I wonder if it would be better that we conditionally set the `StartupWithExtraThreads` here when is_disk_index? This seems require less changes.

Reply:
One thing to think about is tests. We don't necessarily want to create tons of threads for each test that is running parallel.

Reply:
ok. lgtm.

Reply:
More generally, we should not be surprised to find code that isn't concerned, correct, or ideal with in-mem only. It was never the plan to go back to in-mem only once disk index was correct and reasonable.